### PR TITLE
Add travis support

### LINKED ISSUE #20: Add automated build testing with Travis for all images

### PR DESCRIPTION
**Pull request by punkstar**
- Build all images
- Run PHP cli test on all images
- Run tests through web server on Apache images

Closes #20.

### CODE REVIEW

**Comment by punkstar** [comment]
I'm expecting the PHP 7 branches to fail here as we have #24 open at the moment.

File: test-runner.sh [hunk 1]
```diff
@@ -0,0 +1,107 @@
...
+
+################################################################################
+# Run CLI tests on all images
+################################################################################
+echo ""
+echo "Running cli tests:"
+echo ""
+
+docker run --volume $TEST_DIR:/test $IMAGE_NAME php /test/test.php \
+  | grep --context=1000 "$TEST_OK_STRING" || exit 1
```
Comment (tgerulaitis) [inline test-runner.sh:70]: Not sure why the grep context is relevant here?

File: test-runner.sh [hunk 2]
```diff
@@ -0,0 +1,107 @@
...
+cd $PHP_VERSION/$IMAGE_FLAVOUR && docker build -t $IMAGE_NAME . || exit 1
+cd ../..
+
+################################################################################
+# Output the PHP version running in the image
+################################################################################
+echo ""
+echo "Image PHP Version:"
+echo ""
+docker run $IMAGE_NAME php --version || exit 1
```
Comment (tgerulaitis) [inline test-runner.sh:60]: I'm not very familiar with Travis and how it uses instances, but we should probably add `--rm` to `docker run` commands to save resources

**Comment by tgerulaitis** [comment]
:+1:

File: test-runner.sh (hunk 1 above)
Comment (punkstar) [inline test-runner.sh:70]: Add the `--context` so we could see the full output of the `test.php` script, for debugging.

File: test-runner.sh (hunk 2 above)
Comment (punkstar) [inline test-runner.sh:60]: Good shout.

File: test-runner.sh [hunk 3]
```diff
@@ -0,0 +1,107 @@
...
+  # Start the web server
+  docker run -d --volume $TEST_DIR:/var/www/html -p $TEST_PORT:80 $IMAGE_NAME || exit 1
+  echo ""
+
+  # Spent up to a minute trying to connect to the web server
+  max_attempts=12
+  timeout=5
+  attempt=0
+
+  while (( $attempt < $max_attempts ))
```
Comment (punkstar) [inline test-runner.sh:89]: If this loop does actually exceed `$max_attempts` then we still exit with a 0.

File: test-runner.sh (hunk 1 above)
Comment (punkstar) [inline test-runner.sh:70]: Just spoke with @tgerulaitis.  The point is that grep won't return anything if it wasn't a success!  Should `tee` to file and then `grep`.

Comment (punkstar) [inline test-runner.sh:70]: Opened #31.

File: test-runner.sh (hunk 2 above)
Comment (punkstar) [inline test-runner.sh:60]: Opened #29.

File: test-runner.sh (hunk 3 above)
Comment (punkstar) [inline test-runner.sh:89]: Opened #30.

**Pull request merged**